### PR TITLE
refactor: replace DoorItem with DoorAccordion

### PR DESCRIPTION
**Pull request by koonpeng**
## What's new

First try at replacing the current `DoorItem` component with the refactored version from `react-components`.

Some changes are needed to make them play nice:

* For some reason reactour can't correctly capture the component, a workaround is to add a delay before moving to the next step to wait for the animation to finish.
* The components in `react-compoents` have been refactored to include the origin target in their event handlers, e.g. `onDoorControlClick?(event: React.MouseEvent, door: RomiCore.Door, mode: number): void;`. This makes it easier for consumers to do state management as you can now use the same callback for each instance instead of needing to create a new callback for each.
* added a `withSpotlight` HOC that converts a normal `Accordion` to one that supports "spotlight". The HOC exposes a `spotlight` function using the controversal `useImperativeHandle`. Calling the function will scroll the accordion into view and expand it.
  * It is recommended to avoid the use of `useImperativeHandle`, but this is the best solution I can come up with.
* material and react is removed from the dependencies, this is needed to make sure multiple copies of the libraries are not loaded. They are now using the dependencies from the root package, this has some downsides and is the only solution I can find.
  * This causes npm to complain about missing peer dependencies but they are false positives. In the future npm is looking to auto install peer dependencies, we shall see if that change will break monorepos.

## Self-checks

- [x] I'm familiar with and follow this [ Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [x] I added unit-tests for new components
- [x] I tried testing edge cases
- [x] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

**Review by matiasbavera — COMMENTED** [review]
It looks decoupled, thanks for the effort,  I left a couple of comments.

> * It is recommended to avoid the use of `useImperativeHandle`, but this is the best solution I can come up with.

I think it's ok maybe we will be back with another solution after we play around some time with the new structure.

Obs:
Unit tests are failing on the dashboard and on the api-server

File: packages/react-components/package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
   "description": "",
   "main": "lib/index.js",
   "scripts": {
-    "start": "start-storybook -p 6006 -s stories",
+    "preinstall": "node ../../scripts/install.js",
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/react-components/package.json:7]: no, this only installs the root package. The package that requires rmf is `api-server`, and by extension `dashboard` so this remains rmf-free.

File: packages/react-components/stories/door-accordion.stories.tsx [hunk 2]
```diff
@@ -71,6 +70,7 @@ const doorStates = {
 };
 
 export const DoorsPanel: Story = (args) => {
+  console.log(args);
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/react-components/stories/door-accordion.stories.tsx]: fixed

File: packages/dashboard/src/components/dashboard.tsx [hunk 3]
```diff
@@ -465,12 +496,16 @@ export default function Dashboard(_props: {}): React.ReactElement {
...
-                    doorStates={doorStates}
-                    spotlight={doorSpotlight}
-                  />
+                  {doors.map((door) => (
+                    <DoorAccordion
+                      key={door.name}
+                      ref={doorAccordionRefs[door.name].ref}
+                      door={door}
+                      doorState={doorStates[door.name]}
+                      onDoorControlClick={handleOnDoorContolClick}
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/src/components/dashboard.tsx]: fixed

File: packages/dashboard/src/components/tour/tour.tsx [hunk 4]
```diff
@@ -42,10 +39,26 @@ export const DashboardTour = React.memo(
...
 
+    const [update, setUpdate] = React.useState(0);
+
+    /**
+     * Qn: Best solution to handle animations?
+     *
+     * 1. update the tour on an interval.
+     * 2. set a timeout between each type, this has the downside that it would look laggy and
+     * would also breaks when the animation length changes.
+     * 3. Listen for all animation end callback and update the tour. (current solution)
```
Comment (matiasbavera) [inline packages/dashboard/src/components/tour/tour.tsx:53]: Yeah, I think 3 is the best solution for the simple fact that we are relying on events and not just time-related-solutions.

**Review by matiasbavera — APPROVED** [review]
Thanks for this, this will lead us to a good refactor =). LGTM!
